source/posix/condition_variable.c

```diff
@@ -0,0 +1,90 @@
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/condition_variable.h>
+#include <aws/common/mutex.h>
+
+#include <errno.h>
+
+#define NANOS_PER_SEC 1000000000
```

Comment:
should be marked with `ull` suffix otherwise division will be handles as integral and we'll have to handle the overflow/underflow case.

Reply:
good catch.

Reply:
ah, nevermind, these are supposed to be signed AND these are not subject to overflow due to usage.